libr/util/logging.c

```diff
@@ -0,0 +1,127 @@
+/* radare - LGPL - Copyright 2007-2013 - pancake */
+
+#define LOGGING_CONFIGSTR_SIZE 512
+#define LOGGING_OUTPUTBUF_SIZE 512
+
+#include <r_core.h>
+#include <stdarg.h>
+
+// TODO: Use thread-local storage to make these variables thread-safe
+static RLoggingFuncdef cb_logfunc_print = NULL; // Function to call when outputting log string
+static int cfg_loglvl = R_LOGLVL_ERROR; // Logging level output
+static int cfg_logtraplvl = R_LOGLVL_FATALTRAP; // Logging trap level
+static bool cfg_logsrcinfo = false; // Print out debug source info with the output
+static bool cfg_logcolors = false; // Output colored log text based on level
+static char cfg_logfile[LOGGING_CONFIGSTR_SIZE] = ""; // Output text to filename
+static const char *level_tags[] = { // Log level to tag string lookup array
+	[R_LOGLVL_SILLY]     = "SILLY: ",
+	[R_LOGLVL_VERBOSE]   = "VERBOSE: ",
+	[R_LOGLVL_DEBUG]     = "DEBUG: ",
+	[R_LOGLVL_INFO]      = "INFO: ",
+	[R_LOGLVL_WARN]      = "WARNING: ",
+	[R_LOGLVL_ERROR]     = "ERROR: ",
+	[R_LOGLVL_FATAL]     = "FATAL: ",
+	[R_LOGLVL_FATALTRAP] = "FATAL: "
+};
+
+/* TODO: Might want to create a r_logging_use_envvars function so that programs
```

Comment:
I think this could be removed. Programs that do not use core just need to call `r_logging_set_*` functions with the arguments they want and they should be goood to go.

Reply:
Ah yea this comment was before I did the r_logging_set changes. Thanks.